src/array_api_stubs/_draft/statistical_functions.py

```diff
@@ -1,6 +1,66 @@
 from ._types import Optional, Tuple, Union, array, dtype
 
 
+def cumulative_sum(
+    x: array,
+    /,
+    *,
+    axis: Optional[int] = None,
+    dtype: Optional[dtype] = None,
+    startpoint: bool = False,
+    endpoint: bool = True,
+) -> array:
+    """
+    Calculates the cumulative sum of elements in the input array ``x``.
+
+    Parameters
+    ----------
+    x: array
+        input array. Should have a numeric data type.
+    axis: Optional[int]
+        axis along which a cumulative sum must be computed. If ``axis`` is negative, the function must determine the axis along which to compute a cumulative sum by counting from the last dimension.
```

Comment:
It seems to me that this should instead match exactly how the `axis` keyword works in `sum` and other reductions?

Reply:
A few comments. (1) NumPy et al and PyTorch don't support providing multiple axes. So that is already an immediate difference. (2) As mentioned on the [RFC](https://github.com/data-apis/array-api/issues/597#issuecomment-1494778565), flattening behavior is a bit odd and has a less, IMO, clear use case than `sum`, where computing the sum of all elements in an array can make sense, but the cumulative sum, less so.

Reply:
Additionally, for flattening, given that order matters when computing the cumulative sum, I think it is preferable for users to explicitly reshape prior to calling `cumulative_sum`. While row-major order may the desired order, it's not clear that that is an obvious choice in this case. As such, I don't think we want to replicate how the `axis` kwarg works in `sum`.

Reply:
I'll go ahead and mark this as resolved.